### PR TITLE
Phazon Suit Hint in Impact Crater

### PR DESCRIPTION
**Pull request by toasterparty**
Added option to `Hints` tab. By default we are using area name only.

### CODE REVIEW

File: randovania/layout/preset_migration.py [hunk 1]
```diff
@@ -593,6 +593,12 @@ def _migrate_v26(preset: dict) -> dict:
     return preset
 
 
+def _migrate_v27(preset: dict) -> dict:
+    if preset["game"] == "prime1" and "phazon_suit" not in preset["configuration"]["hints"].keys():
```
Comment (henriquegemignani) [inline randovania/layout/preset_migration.py:597]: The second part of the if isn't needed

File: randovania/generator/hint_distributor.py [hunk 2]
```diff
@@ -37,7 +36,6 @@ def _not_empty(it: Iterator) -> bool:
 @dataclasses.dataclass(frozen=True)
 class PreFillParams:
     rng: Random
-    configuration: BaseConfiguration
```
Comment (henriquegemignani) [inline randovania/generator/hint_distributor.py:40]: Any reason why this field is being removed?

File: randovania/games/prime1/hint_distributor.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+from randovania.game_description.hint import (
+    HintLocationPrecision, HintItemPrecision,
+    PrecisionPair
+)
+from randovania.generator.filler.player_state import PlayerState
+from randovania.generator.filler.runner import PlayerPool
+from randovania.generator.hint_distributor import HintDistributor
+
+
+class PrimeHintDistributor(HintDistributor):
```
Comment (henriquegemignani) [inline randovania/games/prime1/hint_distributor.py:13]: Was this class needed right now?

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

File: randovania/layout/preset_migration.py (hunk 1 above)
Comment (toasterparty) [reply randovania/layout/preset_migration.py:597]: I wanted the test rdvgames to use it without having to remake the rdvgame entirely

File: randovania/generator/hint_distributor.py (hunk 2 above)
Comment (toasterparty) [reply randovania/generator/hint_distributor.py:40]: I noticed something wasn't used so I removed it. I can revert this change if desired.

File: randovania/games/prime1/hint_distributor.py (hunk 3 above)
Comment (toasterparty) [reply randovania/games/prime1/hint_distributor.py:13]: No, but I will most likely use it in my next PR. Would you like me to remove it?

Comment (toasterparty) [reply randovania/games/prime1/hint_distributor.py:13]: I was testing around with making other hints (this class works fine), but wanted to get this one hint out in time for the next release.

Comment (henriquegemignani) [reply randovania/games/prime1/hint_distributor.py:13]: If you plan to use then it's fine

File: randovania/generator/hint_distributor.py (hunk 2 above)
Comment (henriquegemignani) [reply randovania/generator/hint_distributor.py:40]: It's more that it was expected to be used, but didn't end up being the case. It's likely to be needed if any game want to configure the hints a bit more though.

Comment (toasterparty) [reply randovania/generator/hint_distributor.py]: reverted

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**